main.rb

```diff
@@ -1,44 +1,250 @@
+    else
+      to_enum(:my_each_with_index)
+    end
+  end
+
+  # 4.my_all?
+  def my_all?(argument = nil)
+    index = 0
+    while index < length
+      if block_given?
+        return false if yield(self[index]) == false
+      else
+        if argument.nil? # when there is no argument
+          return false if self[index].nil? || self[index] == false
+        elsif argument.class == Regexp # when there is an argument
+          return false if self[index].match?(argument) != true
+        elsif argument.class == Class
+          return false if self[index].is_a?(argument) != true
+        else # when the argument is no regexp ni class
+          return false if self[index] != argument
+        end
+      end
+      index += 1
+    end
+    true
+  end
+
+  # 4.my_any?
+  def my_any?(arg = nil)
+    a = 0
+    o = false
+    my = to_a
+    if block_given?
+      while a < my.length
+        if yield(my[a]) == true
+          o = true
+          break
+        end
+        a += 1
+      end
+      o
+    elsif arg.nil?
+      o = true
+    elsif block_given? == true && !arg.nil?
+      o = false
+      while a < my.length
+        if (my[a].is_a? arg) == true
+          o = true
+          break
+        end
+        a += 1
+      end
+      o
+    elsif arg.class == Regexp
+      while a < my.length
+        if !my[a].match(arg).nil?
+          o = true
+          break
+        else
+          o = false
+        end
+        a += 1
+      end
+      o
+    elsif arg.class != Regexp && arg.class == String
+      while a < my.length
+        if my[a] == arg
+          o = true
+          break
+        else
+          o = false
+        end
+        a += 1
+      end
+      o
+    else
+      true
+    end
+  end
+
+  def my_none?(arg = nil)
+    my = to_a
+    a = 0
     if block_given?
-      newArr = []
-      self.my_each {|item| newArr << item if yield(item)}
-      newArr
-    end 
-  end  
+      while a < my.length
+        if yield(my[a]) == true
+          o = false
+          break
+        else
+          a += 1
+          o = true
+        end
+      end
+      o
+    elsif arg.nil?
+      while a < my.length
+        if my[a] == true
+          o = true
+          break
+        else
+          o = false
+        end
+        a += 1
+      end
+      o
+    elsif !arg.nil? && arg.class != Regexp && arg.class != Integer
+      o = true
+      while a < my.length
+        if (my[a].is_a? arg) == true
+          o = false
+          break
+        end
+        a += 1
+      end
+      o
+    elsif arg.class == Regexp
+      while a < my.length
+        if !my[a].match(arg).nil?
+          o = false
+          break
+        else
+          o = true
+        end
+        a += 1
+      end
+      o
+    elsif arg.class != Regexp && !arg.nil?
+      o = true
+      while a < my.length
+        if my[a] == arg
+          o = false
+          break
+        end
+        a += 1
+      end
+      o
+    else
+      true
+    end
+  end
```

Comment:
- [ ] When no block or argument is given, it only returns true if none of the members of the collection is true. 
```ruby
array = [nil, false]
array.my_none? === array.none? => false
```

- [ ] When no block or argument is given, it should return false only if one of the members of the collection is true.

```ruby
array = [nil, false, true, []]
array.my_none? === array.none?
```

**_[OPTIONAL Hint]:_**
- It is possible to get the block of `my_none?` from `my_any?` using `!my_any`. 
```ruby
 def my_none?(arg = nil, &block)
    !my_any?(arg, &block)
  end
```